pkg/controllers/clusterresourcebindingwatcher/watcher_integration_test.go

```diff
@@ -0,0 +1,173 @@
+			labels := crb.GetLabels()
+			labels["test-key"] = "test-value"
+			crb.SetLabels(labels)
+			Expect(k8sClient.Update(ctx, crb)).Should(Succeed(), "failed to update cluster resource binding")
+
+			By("By checking placement controller queue")
+			consistentlyCheckPlacementControllerQueueIsEmpty("watcher should ignore the update labels event and not enqueue the request into the placementController queue")
+		})
+
+		It("Should enqueue the clusterResourcePlacement name for reconciling, when clusterResourceBinding generation changes", func() {
+			Expect(k8sClient.Get(ctx, types.NamespacedName{Name: testCRBName}, crb)).Should(Succeed(), "failed to get cluster resource binding")
+			crb.Spec.State = fleetv1beta1.BindingStateBound
+			Expect(k8sClient.Update(ctx, crb)).Should(Succeed(), "failed to update cluster resource binding")
+
+			By("By checking placement controller queue")
+			eventuallyCheckPlacementControllerQueue(crb.GetLabels()[fleetv1beta1.CRPTrackingLabel], "placementController should receive the cluster resource placement name")
+		})
+
+		It("Should enqueue the clusterResourcePlacement name for reconciling, when clusterResourceBinding status changes - RolloutStarted", func() {
+			updateClusterResourceBindingStatusWithCondition(fleetv1beta1.ResourceBindingRolloutStarted)
+		})
+
+		It("Should enqueue the clusterResourcePlacement name for reconciling, when clusterResourceBinding status changes - Overridden", func() {
+			updateClusterResourceBindingStatusWithCondition(fleetv1beta1.ResourceBindingOverridden)
+		})
+
+		It("Should enqueue the clusterResourcePlacement name for reconciling, when clusterResourceBinding status changes - WorkCreated", func() {
+			updateClusterResourceBindingStatusWithCondition(fleetv1beta1.ResourceBindingWorkCreated)
+		})
+
+		It("Should enqueue the clusterResourcePlacement name for reconciling, when clusterResourceBinding status changes - Applied", func() {
+			updateClusterResourceBindingStatusWithCondition(fleetv1beta1.ResourceBindingApplied)
+		})
+
+		It("Should enqueue the clusterResourcePlacement name for reconciling, when clusterResourceBinding status changes - Available", func() {
+			updateClusterResourceBindingStatusWithCondition(fleetv1beta1.ResourceBindingAvailable)
+		})
+	})
+})
+
+func clusterResourceBindingForTest() *fleetv1beta1.ClusterResourceBinding {
+	return &fleetv1beta1.ClusterResourceBinding{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:   testCRBName,
+			Labels: map[string]string{fleetv1beta1.CRPTrackingLabel: testCRPName},
+		},
+		Spec: fleetv1beta1.ResourceBindingSpec{
+			State:                        fleetv1beta1.BindingStateScheduled,
+			ResourceSnapshotName:         testResourceSnapshotName,
+			SchedulingPolicySnapshotName: testSchedulingPolicySnapshotName,
+			TargetCluster:                testTargetCluster,
+		},
+	}
+}
+
+func updateClusterResourceBindingStatusWithCondition(conditionType fleetv1beta1.ResourceBindingConditionType) {
```

Comment:
```suggestion
func validateWhenUpdateClusterResourceBindingStatusWithCondition(conditionType fleetv1beta1.ResourceBindingConditionType) {
```
rename so that it indicates that you're validating, not just update.